safehttp/mux.go

```diff
@@ -0,0 +1,156 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package safehttp
+
+import (
+	"errors"
+	"net/http"
+)
+
+// TODO: add the missing methods
+const (
+	// HTTP GET request
+	MethodGet = "GET"
+	// HTTP Post request
+	MethodPost = "POST"
+)
+
+// ServeMux TODO
+type ServeMux struct {
+	mux        *http.ServeMux
+	domains    map[string]bool
+	dispatcher Dispatcher
+
+	// Maps user-provided patterns to combined handlers which encapsulate
+	// multiple handlers, each one associated with an HTTP method.
+	handlers map[string]combinedHandler
+}
+
+// NewServeMux returns a safe multiplex containing the list of allowed domains
+// and dispatcher provided by the user.
+func NewServeMux(dispatcher Dispatcher, domains ...string) *ServeMux {
+	d := map[string]bool{}
+	for _, host := range domains {
+		d[host] = true
+	}
+	return &ServeMux{mux: http.NewServeMux(), domains: d, dispatcher: dispatcher, handlers: map[string]combinedHandler{}}
```

Comment:
Nit: break in multiple lines.

Reply:
Line breaks are not strictly necessary in Go code, but this seems like a case where it'd benefit readability.